### PR TITLE
Update EnableEpochsHandler interface

### PR DESCRIPTION
**Pull request by sstanculeanu**

### CODE REVIEW

File: builtInFunctions/esdtDataStorage_test.go [hunk 1]
```diff
@@ -621,7 +630,9 @@ func TestEsdtDataStorage_WasAlreadySentToDestinationShard(t *testing.T) {
 	assert.False(t, val)
 	assert.Nil(t, err)
 
-	enableEpochsHandler.IsSendAlwaysFlagEnabledField = false
+	enableEpochsHandler.IsFlagEnabledCalled = func(flag core.EnableEpochFlag) bool {
+		return flag == SaveToSystemAccountFlag
```
Comment (iulianpascalau) [inline builtInFunctions/esdtDataStorage_test.go:634]: original code was:
```
enableEpochsHandler.IsSendAlwaysFlagEnabledField = false
```
this works but the flag set is unrelated

File: builtInFunctions/esdtDataStorage_test.go [hunk 2]
```diff
@@ -643,15 +654,21 @@ func TestEsdtDataStorage_SaveNFTMetaDataToSystemAccount(t *testing.T) {
...
-	enableEpochsHandler.IsSaveToSystemAccountFlagEnabledField = false
+	enableEpochsHandler.IsFlagEnabledCalled = func(flag core.EnableEpochFlag) bool {
+		return false
+	}
 	err := e.SaveNFTMetaDataToSystemAccount(nil)
 	assert.Nil(t, err)
 
-	enableEpochsHandler.IsSaveToSystemAccountFlagEnabledField = true
+	enableEpochsHandler.IsFlagEnabledCalled = func(flag core.EnableEpochFlag) bool {
+		return flag == SaveToSystemAccountFlag || flag == SendAlwaysFlag
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply builtInFunctions/esdtDataStorage_test.go:664]: same here.. when args are created, both flags are enabled.. the issue is that on L658 I returned false, not true for flag SendAlwaysFlag.. pushed this fix

File: builtInFunctions/esdtDataStorage_test.go [hunk 3]
```diff
@@ -643,15 +654,21 @@ func TestEsdtDataStorage_SaveNFTMetaDataToSystemAccount(t *testing.T) {
...
+	}
 	err = e.SaveNFTMetaDataToSystemAccount(nil)
 	assert.Nil(t, err)
 
-	enableEpochsHandler.IsSendAlwaysFlagEnabledField = false
+	enableEpochsHandler.IsFlagEnabledCalled = func(flag core.EnableEpochFlag) bool {
+		return flag == SaveToSystemAccountFlag
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply builtInFunctions/esdtDataStorage_test.go:670]: same here, SaveToSystemAccountFlag should stay active

File: builtInFunctions/baseActiveHandler_test.go [hunk 4]
```diff
@@ -24,11 +25,15 @@ func TestBaseActiveHandler_IsActive(t *testing.T) {
 
 	enableEpochsHandler := mock.EnableEpochsHandlerStub{}
 	handler = &baseActiveHandler{
-		activeHandler: enableEpochsHandler.IsSCDeployFlagEnabled,
+		activeHandler: func() bool {
+			return enableEpochsHandler.IsFlagEnabled(SetGuardianFlag)
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply builtInFunctions/baseActiveHandler_test.go:29]: completely removed, as this is already done on L13-L22

**Review by iulianpascalau — DISMISSED** [review]

File: builtInFunctions/multiESDTNFTTransfer.go [hunk 5]
```diff
@@ -308,7 +311,8 @@ func (e *esdtNFTMultiTransfer) processESDTNFTMultiTransferOnSenderShard(
 	topicTokenData := make([]*TopicTokenData, 0)
 	for i := uint64(0); i < numOfTransfers; i++ {
 		tokenStartIndex := startIndex + i*argumentsPerTransfer
-		if len(vmInput.Arguments[tokenStartIndex+2]) > core.MaxLenForESDTIssueMint && e.enableEpochsHandler.IsConsistentTokensValuesLengthCheckEnabled() {
+		isConsistenTokensValuesLenghtCheckEnabled := e.enableEpochsHandler.IsFlagEnabled(ConsistentTokensValuesLengthCheckFlag)
```
Comment (bogdan-rosianu) [inline builtInFunctions/multiESDTNFTTransfer.go:314]: can be extracted outside the for loop. Also, typo in variable's name

Comment (sstanculeanu) [reply builtInFunctions/multiESDTNFTTransfer.go]: pushed

**Review by bogdan-rosianu — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**